legend-engine-xt-nonrelationalStore-mongodb-executionPlan-test/src/main/java/org/finos/legend/engine/plan/execution/stores/mongodb/test/shared/MongoDBCommands.java

```diff
@@ -0,0 +1,136 @@
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class MongoDBCommands
+{
+
+    public static final String DB_ROOT_USERNAME = "sa";
+    public static final String DB_ROOT_PASSWORD = "sa";
+
+    private static final String DB_USER_DATABASE = "userDatabase";
+    private static final String DB_USER_DB_PERSON_COLLECTION = "person";
+    private static final int MONGO_PORT = 27017;
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoDBCommands.class);
+    public static final Map<String, GenericContainer<MongoDBContainer>> CONTAINERS = Maps.mutable.empty();
+    public static final ObjectMapper OBJECT_MAPPER = ObjectMapperFactory.getNewStandardObjectMapperWithPureProtocolExtensionSupports();
+
+    public static String START_SERVER_FUNCTION = "startMongoDBTestServer_String_1__URL_1_";
+
+    public static Root_meta_pure_functions_io_http_URL startServer(String imageTag)
+    {
+        LOGGER.debug("Starting MongoDB docker image + " + imageTag);
+
+        Root_meta_pure_functions_io_http_URL_Impl url = new Root_meta_pure_functions_io_http_URL_Impl("mongoDBUrl");
+        GenericContainer<MongoDBContainer> container = CONTAINERS.computeIfAbsent(imageTag, MongoDBCommands::createContainer);
+        url._host(container.getHost());
+        url._port(container.getMappedPort(MONGO_PORT));
+        url._path("/");
+
+        return url;
+    }
+
+    public static String STOP_SERVER_FUNCTION = "stopMongoDBTestServer_String_1__Nil_0_";
+
+    public static void stopServer(String imageTag)
+    {
+        LOGGER.debug("Stopping MongoDB docker image + " + imageTag);
+        Optional.ofNullable(CONTAINERS.remove(imageTag)).ifPresent(GenericContainer::stop);
+    }
+
+
+    public static String REQUEST_SERVER_FUNCTION = "requestMongoDBTestServer_String_1__String_1__String_1_";
+
+    public static String request(String imageTag, String json)
+    {
+
+        LOGGER.debug("Performing MongoDB request for image: " + imageTag);
+        Integer mongoPort = getPortForRunningContainerImage(imageTag);
+
+        MongoClient adminClient = mongoClientForRootAdminWithStaticUserNamePassword(mongoPort);
+
+        MongoDatabase userDatabase = adminClient.getDatabase(DB_USER_DATABASE);
+
+        Document bsonCmd = Document.parse(json);
+        Document document = userDatabase.runCommand(bsonCmd);
+
+        String jsonResult = document.toJson();
+        LOGGER.debug("MongoDB request result:\n" + jsonResult);
+
+        return jsonResult;
+    }
+
+    private static Integer getPortForRunningContainerImage(String imageTag)
+    {
+        return CONTAINERS.get(imageTag).getMappedPort(MONGO_PORT);
+    }
+
+
+    private static GenericContainer<MongoDBContainer> createContainer(String imageTag)
+    {
+
+        GenericContainer<MongoDBContainer> container =
```

Comment:
Did we consider using this rather than GenericContainer? https://www.testcontainers.org/modules/databases/mongodb/

Should not change much, but could make things clearer and we could get some default things done.

Reply:
Yes, could not use MongoDBContainer as is because there is a gap when we add a custom root user/pass there is a thread here explaining. We may be able to use a different work around as I agrtee it would be cleaner to not use the GenericContainer here ( will address this in next PR) https://github.com/testcontainers/testcontainers-java/issues/4695#issuecomment-1350366355

 